gateway/handler.go

```diff
@@ -171,6 +172,29 @@ func (i *handler) optionsHandler(w http.ResponseWriter, r *http.Request) {
 	i.addUserHeaders(w) // return all custom headers (including CORS ones, if set)
 }
 
+type requestData struct {
+	// Defined for all requests.
+	begin          time.Time
+	logger         *zap.SugaredLogger
+	contentPath    ipath.Path
+	responseFormat string
+	responseParams map[string]string
+
+	// Defined for non IPNS Record requests.
+	immutablePath ImmutablePath
+
+	// Defined if resolution has already happened.
+	pathMetadata *ContentPathMetadata
+	resolvedPath *ImmutablePath
+}
+
+func (rq *requestData) maybeResolvedPath() ImmutablePath {
```

Comment:
Would this be a better description of that this does?
I really want to avoid `maybe` as is pretty ambiguous :dizzy:  

```suggestion
// mostlyResolvedPath is an opportunistic optimization that returns the mostly resolved version of ImmutablePath available.
// It does not guarantee it is fully resolved, nor that it is the original.
func (rq *requestData) mostlyResolvedPath() ImmutablePath {
```